### PR TITLE
Improve horizontal scrolling in grid

### PR DESCRIPTION
**Pull request by jpzwarte**

### CODE REVIEW

**Review by anna-lach — APPROVED** [review]

File: packages/components/grid/src/grid.scss [hunk 1]
```diff
@@ -143,7 +228,9 @@ tr {
 th,
 td {
   align-items: center;
-  background: var(--_vertical-border), var(--_cell-background);
+
+  // background: var(--_vertical-border), var(--_cell-background);
+  background: var(--_cell-background);
```
Comment (Diaan) [inline packages/components/grid/src/grid.scss:233]: comment

File: packages/components/grid/src/grid.ts [hunk 2]
```diff
@@ -304,12 +306,23 @@ export class Grid<T = any> extends ScopedElementsMixin(LitElement) {
...
-        <tbody @visibilityChanged=${this.#onVisibilityChanged} part="tbody">
+        <tbody @visibilityChanged=${this.#onVisibilityChanged} id="tbody" part="tbody">
           ${virtualize({
             items: this.view.rows,
             renderItem: (item, index) => this.renderItem(item, index)
           })}
         </tbody>
+        ${this.scrollbar
+          ? html`
+              <tfoot>
```
Comment (Diaan) [inline packages/components/grid/src/grid.ts:317]: is the `tfoot` the right element for this? (although i don't know what element would fit better semantically 😅 )

File: packages/components/grid/src/grid.ts [hunk 3]
```diff
@@ -457,7 +470,7 @@ export class Grid<T = any> extends ScopedElementsMixin(LitElement) {
       .filter(col => !col.hidden && col.autoWidth)
       .forEach(col => {
         const index = this.view.headerRows[this.view.headerRows.length - 1].indexOf(col),
-          cells = this.renderRoot.querySelectorAll<HTMLElement>(`:where(td, th):nth-child(${index + 1})`);
+          cells = this.renderRoot.querySelectorAll<HTMLElement>(`:where(tbody td, thead th):nth-child(${index + 1})`);
```
Comment (Diaan) [inline packages/components/grid/src/grid.ts:473]: what about th cells in the tbody, when you have a "heading" in the first column in the body for example.

File: packages/components/grid/src/grid.scss [hunk 4]
```diff
@@ -42,55 +45,137 @@
...
+    }
+  }
+}
+
+:host([scrollable-start]) {
+  --_vertical-border-end: linear-gradient(var(--_border-color), var(--_border-color)) 100% 100% / var(--_border-width)
+    100% no-repeat;
+
+  table:not(:has(.sticky-start-last))::before {
+    background: linear-gradient(to right, rgb(79 79 79 / 10%), transparent);
```
[the remark this answers is not in the document]
Comment (jpzwarte) [reply packages/components/grid/src/grid.scss:66]: Not yet; Roald was going to do this.

File: packages/components/grid/src/grid.ts (hunk 2 above)
Comment (jpzwarte) [reply packages/components/grid/src/grid.ts:317]: `<tfoot>` is the only element that is acceptable: the HTML parser is strict when it comes to allowed elements inside of `<table>`. So this is a necessary "hack".

File: packages/components/grid/src/grid.ts (hunk 3 above)
Comment (jpzwarte) [reply packages/components/grid/src/grid.ts:473]: Is that correct HTML? I thought `<th>` elements are only allowed in the `<thead>` element?

Comment (jpzwarte) [reply packages/components/grid/src/grid.ts]: Changed

File: packages/components/grid/src/grid.scss (hunk 4 above)
[the remark this answers is not in the document]
Comment (jpzwarte) [reply packages/components/grid/src/grid.scss:66]: But this is most likely happening on a different branch. We need to refactor the styling of the grid anyway to use the new tokens. So i want to keep it like this for now.

**Review by Diaan — APPROVED** [review]

**Pull request merged**